text/java/0012-opentelemetry.md

```diff
@@ -0,0 +1,44 @@
+# OpenTelemetry Support
+
+## Summary
+
+The [Paketo Java Buildpack](https://github.com/paketo-buildpacks/java) currently supports the APM tools Google Stackdriver, Azure Application Insights, and DataDog. This RFC proposes we add support for the open-source OpenTelemetry tool as well.
+
+## Motivation
+
+OpenTelemetry has quickly became a standard for instrumenting, collecting, and exporting telemetry data. It's adopted and supported by industry leaders in the observability space. For Java applications, an OpenTelemetry Java Agent is available for instrumenting the code and exporting traces, metrics, and logs.
+
+## Detailed Explanation
+
+This RFC proposes the following changes:
+
+- Create a new buildpack, `paketo-buildpacks/opentelemetry`.
+- The buildpack will install the OpenTelemetry Agent, which includes everything needed to instrument and export telemetry data.
+- The buildpack will adjust the JVM arguments to enable the OpenTelemetry Agent.
+- Users would configure the OpenTelemetry Agent using the [standard environment variables](https://opentelemetry.io/docs/instrumentation/java/automatic/agent-config/#configuring-the-agent) provided by the OpenTelemetry project itself.
```

Comment:
That's a very good point, thanks. I can work on that.

Reply:
For what it's worth, we did not do this with Datadog but the Java agent does not require any sensitive data to be passed into it. 

The helper with Azure App Insights is a better example of how you'd typically do this though: https://github.com/paketo-buildpacks/azure-application-insights/blob/main/helper/properties.go.  It finds the bindings and turns them into environment variables in the helper. Since the helper runs just prior to the app starting up, the environment variables are not saved with the image.

If you need to write a config file, make sure to write it to `/tmp` as that will be the only location in some environments where you can write. It also tends to (always?) be configured such that it's thrown away after the container exits which is a little safer.

Reply:
@dmikusa-pivotal thanks for the tip! I'm working on it and I'll update PR and Buildpack repo in the next few days.

Reply:
I have updated the RFC document with the new information on using the configuration file. I'm still working on the implementation on this branch: https://github.com/ThomasVitale/buildpacks-opentelemetry/tree/support-config-file